libraries/UR_Atmosphere/UR_Atmosphere.cpp

```diff
@@ -0,0 +1,222 @@
+        pQFE = pressure_qnh;
+        break;
+    case INCH_MERCURY:                      // [inHg]   inches mercury
+        pQFE = pressure_qnh / ChPaTOinHG;
+        break;
+    case MILLIMETER_MERCURY:                // [mmHg]   millimiters mercury
+        pQFE = pressure_qnh / ChPaTOmmHG;
+        break;
+    default:
+        break;
+    }
+
+    // Calculation(s)
+    pressure_qnh = p0hPa * powf(powf((pQFE / p0hPa), (-1 * (CRGasSI * dTdh0SI)) / CgSI) - (hQFE * dTdh0SI) / (Temp0 + temp - 15), (-1 * CgSI) / (CRGasSI * dTdh0SI));
+
+    // Unit Conversion(s)
+    switch (resultQNHUnit) {
+    case HECTO_PASCAL:                              // [hPa]    hectoPascal
+        break;
+    case INCH_MERCURY:                              // [inHg]   inches mercury
+        pressure_qnh = pressure_qnh * ChPaTOinHG;
+        break;
+    case MILLIMETER_MERCURY:                        // [mmHg]   millimiters mercury
+        pressure_qnh = pressure_qnh * ChPaTOmmHG;
+        break;
+    default:
+        break;
+    }
+
+    const float resultQNH = pressure_qnh;
+    return resultQNH;
+}
+
+// Calculate QFE by hPa and feet by default.
+// [1] Meteorology Specification A2669 Version 4.00 Page 100
+float UR_Atmosphere::calculate_qfe(float alt_qfe, float pressure_qfe, float temp, altitude_unit_t alt_qfe_unit) const
+{
+    uint8_t pQFEUnit = PRESSURE_UNIT::HECTO_PASCAL;
+    uint8_t resultQFEUnit = PRESSURE_UNIT::HECTO_PASCAL;
+    float hQNH = 0;
+    float pQNH = 0;
+
+    switch (alt_qfe_unit) {
+    case FEET:                                 // [ft] feet
+        hQNH = alt_qfe * CftTOm;
+        break;
+    case METER:                                 // [m]  meters
+        hQNH = alt_qfe;
+        break;
+    default:
+        break;
+    }
+
+    switch (pQFEUnit) {
+    case HECTO_PASCAL:                      // [hPa]    hectoPascal
+        pQNH = pressure_qfe;
+        break;
+    case INCH_MERCURY:                      // [inHg]   inches mercury
+        pQNH = pressure_qfe / ChPaTOinHG;
+        break;
+    case MILLIMETER_MERCURY:                // [mmHg]   millimiters mercury
+        pQNH = pressure_qfe / ChPaTOmmHG;
+        break;
+    default:
+        break;
+    }
+
+    // Calculations
+    alt_qfe = p0hPa * powf(powf((pQNH / p0hPa), (-1 * (CRGasSI * dTdh0SI)) / CgSI) + ((hQNH * dTdh0SI) / (Temp0 + temp - 15)), (-1 * CgSI) / (CRGasSI * dTdh0SI));
+
+    // Unit Conversions
+    switch (resultQFEUnit) {
+    case HECTO_PASCAL:                      // [hPa]    hectoPascal
+        break;
+    case INCH_MERCURY:                      // [inHg]   inches mercury
+        alt_qfe = alt_qfe * ChPaTOinHG;
+        break;
+    case MILLIMETER_MERCURY:                // [mmHg]   millimiters mercury
+        alt_qfe = alt_qfe * ChPaTOmmHG;
+        break;
+    default:
+        break;
+    }
+
+    const float resultQFE = alt_qfe;
+    return resultQFE;
+}
+
+void UR_Atmosphere::consume_updated(void)
```

Comment:
There is no point to this method, you always call it, before then calling update.